ntp-proto/src/peer.rs

```diff
@@ -21,8 +20,8 @@ pub enum NtsError {
 
 pub struct PeerNtsData {
     pub(crate) cookies: CookieStash,
-    pub(crate) c2s: Aes128SivAead,
-    pub(crate) s2c: Aes128SivAead,
+    pub(crate) c2s: Box<dyn Cipher>,
+    pub(crate) s2c: Box<dyn Cipher>,
```

Comment:
so, we use `Box<dyn Cipher>` here to support multiple ciphers, that possibly use different sizes for their keys. Additionally the keys are pretty big so storing them on the heap to keep `PeerNtsData` small is pretty reasonable.

maybe add a comment here (to me Box<dyn T> is a bit of an exotic feature)

Reply:
Done